### DISCUSSION TITLE
Support scanning all commits in git repos [trivy repo]

### DISCUSSION THREAD

**Discussion by unnirapido** [aquasecurity/trivy#5341] [Ideas]
### Description

I believe the `trivy repo` can only detect secrets in the particular commit or branch we provide as arg. This feature will resemble gitleaks, where trivy can scan all the commits in the particular git repo.
1. Is this already supported by trivy?
2. Is this feature already in the future plan? (I'm happy to contribute towards this feature)

### Target

Git Repository

### Scanner

Secret

**Comment by DmitriyLewen** [comment]
Hello @unnirapido 

> Is this already supported by trivy?

Trivy doesn't currently support scan all commits of repo.

> Is this feature already in the future plan? (I'm happy to contribute towards this feature)

I'm worried about repositories with a lot of commits.
What about memory costs and running time?
It might make sense to scan a range of commits (use start and end commits)
@knqyf263 wdyt?
